lib/internal/event_target.js

```diff
@@ -190,6 +192,23 @@ class NodeCustomEvent extends Event {
     }
   }
 }
+
+// Weak listener cleanup
+// This has to be lazy for snapshots to work
+let weakListenersState = null;
+// The resource needs to retain the callback so that it doesn't
+// get garbage collected now that it's weak.
+let objectToWeakListenerMap = null;
+function weakListeners() {
+  if (!weakListenersState) {
+    weakListenersState = new globalThis.FinalizationRegistry(
+      (listener) => listener.remove()
+    );
+    objectToWeakListenerMap = new SafeWeakMap();
+  }
```

Comment:
```suggestion
  weakListenersState ??= new globalThis.FinalizationRegistry(
      (listener) => listener.remove()
  );
  objectToWeakListenerMap ??= new SafeWeakMap();
```

N.B.: Null coalescing assignment is not available on Node.js 14, so ignore this suggestion if we want to backport this PR to v14 in the future.

Reply:
Hmm, it looks like `FinalizationRegistry` _is_ available in Node.js 14 but the chances for backporting EventTarget are pretty low since it introduces a new global.

I'll go ahead and apply this.